frontend/dds_mvp.js

```diff
@@ -145,13 +153,46 @@ function inputIsValid(hands) {
 
         for (const card of hand) {
             const pip = card.substring(1);
+
             if (!PIPS.includes(pip)) {
                 return "Please use only these pips: " + PIPS;
             }
+
+            if (deck[card]) {
+                if (deck[card] == 1) {
+                    duplicates.push(card);
+                }
+
+                deck[card]++;
+            } else {
+                deck[card] = 1;
+            }
         }
     }
 
-    return "";
+    if (duplicates.length) {
+        var error_message = "Duplicated card";
+
+        if (duplicates.length > 1) {
+            error_message += "s";
+        }
+
+        error_message += ": ";
+
+        for (const card of duplicates) {
+            const suit_letter = card.substring(0, 1);
+            const pip = card.substring(1);
+            const suit_symbol = SUIT_SYMBOLS[suit_letter];
+
+            error_message += suit_symbol;
+            error_message += pip;
+            error_message += " ";
+        }
+
+        return error_message;
+    } else {
```

Comment:
This else looks redundant. But there is many ways to go about removing it. First is just removing it because previous if branch is an error check and returns in the end. This works fine if we want code reporting only an error.

Often it is more user friendly to report all errors which can be detected same time. Idea is prominently used in compilers which parsing after first error to report all errors same time. If this idea is prefered then a better solution would be to defined error_message as empty list in begin of funcion. Then each error check just appends to the error_message. Then return statement would just join the list to a string with a line separation tag(s) in between of different error messages (tag could be `<br>` or `</div><div class=errormsg>` or `</errormsg><errormsg>`).

Current development situation indicates the former solution seems like a better solution for now. This is just an experimental prototype so no need to go all out designing all features from start.

Reply:
Removed the else per your suggestion.

After we pick a JS library we should choose a set of style guidelines. Is there one you like? Please reply on the Discord since I'm about to merge this pull.